File: BDD.Practices/src/main/java/pageObjects/RegistrationPage.java
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import Base.BaseClass;

public class RegistrationPage extends BaseClass {

	public WebDriver driver;

	public RegistrationPage(WebDriver driver) {
		this.driver = driver;
		PageFactory.initElements(driver, this);
	}

	@FindBy(className = "page-heading")
	WebElement createAcountLabel;

	@FindBy(id = "id_gender1")
	WebElement maleGender;

	@FindBy(id = "id_gender2")
	WebElement femaleGender;

	@FindBy(id = "customer_firstname")
	WebElement firstNameTextBox;

	@FindBy(id = "customer_lastname")
	WebElement lastNameTextBox;

	@FindBy(id = "passwd")
	WebElement passwordTextBox;

	@FindBy(id = "address1")
	WebElement addressLine1;

	@FindBy(id = "city")
	WebElement city;

	@FindBy(id = "id_state")
	WebElement stateDropDown;

	@FindBy(id = "postcode")
	WebElement zipcode;

	@FindBy(id = "id_country")
	WebElement countryDropdown;

	@FindBy(id = "other")
	WebElement additionalInfo;

	@FindBy(id = "phone_mobile")
	WebElement mobileNo;

	@FindBy(id = "submitAccount")
	WebElement registerButton;

	//Actions to be performed

	public void selectGenderTitle(String title) {
		waitForElement(driver, createAcountLabel, 15);
		if (title.equals("Mr."))
			maleGender.click();
		else
			femaleGender.click();
	}

	public void enterFirstName(String fname) {
		enterText(firstNameTextBox, fname);
	}

	public void enterLastName(String fname) {
		enterText(lastNameTextBox, fname);
	}

	public void enterPasswordForRegistration(String pwd) {
		enterText(passwordTextBox, pwd);
	}

	public void enterAddress(String address) {
		enterText(addressLine1, address);
	}

	public void enterCity(String cityName) {
		enterText(city, cityName);
	}

	public void selectState(String state) {
		selectDropDownByText(driver, stateDropDown, state);
	}

	public void enterZipCode(String zipc) {
		enterText(zipcode, zipc);
	}

	public void selectCountry(String country) {
		selectDropDownByText(driver, countryDropdown, country);
	}

	public void enterAdditionalInfo(String no) {
		enterText(additionalInfo, no);
	}

	public void enterMobileNo(String no) {
		enterText(mobileNo, no);
	}

	public void clickOnRegisterButton() {
		registerButton.click();
	}

}

File: WDIO-TESTING/pageobjects/registration.page.js
import Page from './page';

/**
 * sub page containing specific selectors and methods for a specific page
 */
class RegistrationPage extends Page{
    get appHeader () {return $('#header_logo')}
    get registrationPageHeading () { return $('.page-heading') }
    get maleGender () { return $('#id_gender1') }
    get femaleGender () { return $('#id_gender2') }
    get firstNameTextBox () { return $('#customer_firstname') }
    get lastNameTextBox () { return $('#customer_lastname') }
    get passwordTextBox () { return $('#passwd') }
    get addressLine1 () { return $('#address1') }
    get cityTextBox () { return $('#city') }
    get stateDropDown () { return $('#id_state') }
    get zipcode () { return $('#postcode') }
    get countryDropdown () { return $('#id_country') }
    get additionalInfo () { return $('#other') }
    get mobileNo () { return $('#phone_mobile') }
    get registerButton () { return $('#submitAccount') }
    get headerUserInfo () { return $$('a.account span') }

    
    async selectGenderTitle(title) {
        await this.appHeader;
        if (title === ("Mr.")){ 
            await (await this.maleGender).click(); 
        }
		else { 
            await (await this.femaleGender).click()
        }
    }
    async enterFirstName (fname) { 
        await (await this.firstNameTextBox).setValue(fname)
    }
    async enterLastName (lname) { 
        await (await this.lastNameTextBox).setValue(lname) 
    }
    async enterPassword (password) {
        await (await this.passwordTextBox).setValue(password)
    }
    async enterAddress (address) { 
        await (await this.addressLine1).setValue(address) 
    }
    async enterCity (city) { 
        await (await this.cityTextBox).setValue(city) 
    }
    async selectState (stateName) { 
        await (await this.stateDropDown).selectByVisibleText(stateName) 
    }
    async enterZipcode (zipcod) { 
        await (await this.zipcode).setValue(zipcod) 
    }
    async selectCountry (countryName) { 
        await (await this.countryDropdown).selectByVisibleText(countryName) 
    }
    async enterAdditionalInfor (addinformation) { 
        await (await this.additionalInfo).setValue(addinformation) 
    }
    async enterMobileno (mobileno) { 
        await (await this.mobileNo).setValue(mobileno) 
    }
    async clickOnRegisterButton () { 
        await (await this.registerButton).click() 
    }   

    open () {
        return super.open();
    }

}

export default new RegistrationPage();